src/SFML/Graphics/Texture.cpp

```diff
@@ -259,55 +260,50 @@ bool Texture::loadFromStream(InputStream& stream, const IntRect& area)
 bool Texture::loadFromImage(const Image& image, const IntRect& area)
 {
     // Retrieve the image size
-    int width  = static_cast<int>(image.getSize().x);
-    int height = static_cast<int>(image.getSize().y);
+    Vector2i size(image.getSize());
+
+    IntRect absArea(area.getAbsoluteRect());
 
     // Load the entire image if the source area is either empty or contains the whole image
-    if (area.width == 0 || (area.height == 0) ||
-        ((area.left <= 0) && (area.top <= 0) && (area.width >= width) && (area.height >= height)))
+    if (absArea.size.x == 0 || absArea.size.y == 0 || absArea.contains(IntRect({0, 0}, size)))
     {
-        // Load the entire image
-        if (create(image.getSize()))
+        if (!create(image.getSize()))
```

Comment:
Why have we inverted the bool condition for this? I'm fairly sure if create fails we shouldn't return true

Reply:
That's an actual mistake, thank you for pointing out!